libraries/ESP8266WiFiMesh/src/ESP8266WiFiMesh.h

```diff
@@ -24,52 +24,193 @@
+   * 
+   */
+  ESP8266WiFiMesh(uint32_t chip_id, std::function<String(String)> handler);
+
+  /**
+   * Scan for other nodes, and exchange the chosen message with any that are found.
+   *
+   * @message The message to send to all other nodes.
+   * 
+   */
+  void attemptScan(String message);
+  
+  ////////////////////////////</DEPRECATED> TODO: REMOVE IN 2.5.0////////////////////////////
+
+  /**
+   * WiFiMesh Constructor method. Creates a WiFi Mesh Node, ready to be initialised.
+   *
+   * @param requestHandler The callback handler for dealing with received requests. Takes a string as an argument which
+   *          is the request string received from another node and returns the string to send back.
+   * @param responseHandler The callback handler for dealing with received responses. Takes a string as an argument which
+   *          is the response string received from another node. Returns a transmission status code as a transmission_status_t.
+   * @param networkFilter The callback handler for deciding which WiFi networks to connect to.
+   * @param mesh_password The WiFi password for the mesh network.
+   * @param mesh_name The name of the mesh network. Used as prefix for the node SSID and to find other network nodes in the example network filter function.
+   * @param node_id The id for this mesh node. Used as suffix for the node SSID. If set to "", the id will default to ESP.getChipId().
+   * @param verbose_mode Determines if we should print the events occurring in the library to Serial. Off by default.
+   * @param mesh_wifi_channel The WiFi channel used by the mesh network. Valid values are integers from 1 to 13. Defaults to 1.
+   *                    WARNING: The ESP8266 has only one WiFi channel, and the the station/client mode is always prioritized for channel selection.
+   *                    This can cause problems if several ESP8266WiFiMesh instances exist on the same ESP8266 and use different WiFi channels. 
+   *                    In such a case, whenever the station of one ESP8266WiFiMesh instance connects to an AP, it will silently force the 
+   *                    WiFi channel of any active AP on the ESP8266 to match that of the station. This will cause disconnects and possibly 
+   *                    make it impossible for other stations to detect the APs whose WiFi channels have changed.
+   * @param server_port The server port used by the AP of the ESP8266WiFiMesh instance. If multiple APs exist on a single ESP8266, each requires a separate server port. 
+   *              If two AP:s on the same ESP8266 are using the same server port, you must call deactivateAP on the active AP before calling activateAP on the inactive AP.                   
+   * 
+   */
+  ESP8266WiFiMesh(std::function<String(String, ESP8266WiFiMesh *)> requestHandler, std::function<transmission_status_t(String, ESP8266WiFiMesh *)> responseHandler, 
```

Comment:
This is precisely where separation of the arg types would be most useful. As it is, the declaration is very difficult to read.

Reply:
Use references instead of value and pointer as needed